src/analyze/analyze.c

```diff
@@ -1139,22 +1147,49 @@ static int graph_one(sd_bus *bus, const UnitInfo *u, char *patterns[], char *fro
         assert(u);
 
         if (IN_SET(arg_dot, DEP_ORDER, DEP_ALL)) {
-                r = graph_one_property(bus, u, "After", "green", patterns, from_patterns, to_patterns);
+                r = graph_one_property(bus, u, "After", dependency_color[UNIT_AFTER],
+                                                        dependency_style[UNIT_AFTER], patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
         }
 
         if (IN_SET(arg_dot, DEP_REQUIRE, DEP_ALL)) {
-                r = graph_one_property(bus, u, "Requires", "black", patterns, from_patterns, to_patterns);
+                r = graph_one_property(bus, u, "Requires", dependency_color[UNIT_REQUIRES],
+                                                           dependency_style[UNIT_REQUIRES], patterns, from_patterns, to_patterns);
+                if (r < 0)
+                        return r;
+                r = graph_one_property(bus, u, "Requisite", dependency_color[UNIT_REQUISITE],
+                                                            dependency_style[UNIT_REQUISITE], patterns, from_patterns, to_patterns);
+                if (r < 0)
+                        return r;
+                r = graph_one_property(bus, u, "Wants", dependency_color[UNIT_WANTS],
+                                                        dependency_style[UNIT_WANTS], patterns, from_patterns, to_patterns);
+                if (r < 0)
+                        return r;
+                r = graph_one_property(bus, u, "Conflicts", dependency_color[UNIT_CONFLICTS],
+                                                            dependency_style[UNIT_CONFLICTS], patterns, from_patterns, to_patterns);
+                if (r < 0)
+                        return r;
+                r = graph_one_property(bus, u, "BindsTo", dependency_color[UNIT_BINDS_TO],
+                                                          dependency_style[UNIT_BINDS_TO], patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
-                r = graph_one_property(bus, u, "Requisite", "darkblue", patterns, from_patterns, to_patterns);
+                r = graph_one_property(bus, u, "PartOf", dependency_color[UNIT_PART_OF],
+                                                         dependency_style[UNIT_PART_OF], patterns, from_patterns, to_patterns);
                 if (r < 0)
                         return r;
-                r = graph_one_property(bus, u, "Wants", "grey66", patterns, from_patterns, to_patterns);
```

Comment:
this can be done in a loop now I figure iterating through the enum, and using `unit_dependency_to_string()`

Reply:
Now, there's a loop for UNIT_REQUIRES until UNIT_PART_OF.